File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var divideArray = function(nums) {
    return Object.entries(nums.reduce((acc, val) => ({...acc, [val]: acc[val] + 1 || 1}), {})).every(([,x]) => x % 2 === 0)
};

File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.kt
class Solution {
    fun divideArray(nums: IntArray): Boolean {
        val cnt = IntArray(501){0}
        for (n in nums) {
            cnt[n]++
        }
        for (n in cnt) {
            if (n % 2 !== 0) {
                return false
            }
        }
        return true
    }
}